src/provider/systemdmanager/systemdunit.rs

```diff
@@ -131,38 +267,32 @@ impl SystemDUnit {
     }
 
     /// Add a key=value entry to the specified section
-    fn add_property(&mut self, section: &'static str, key: &str, value: &str) {
-        let section = self
-            .sections
-            .entry(String::from(section))
-            .or_insert_with(HashMap::new);
+    fn add_property(&mut self, section: Section, key: &str, value: &str) {
+        let section = self.sections.entry(section).or_insert_with(HashMap::new);
         section.insert(String::from(key), String::from(value));
     }
 
-    fn add_env_var(&mut self, name: &str, value: &str) {
-        self.environment
-            .insert(String::from(name), String::from(value));
-    }
-
     /// Retrieve content of the unit file as it should be written to disk
     pub fn get_unit_file_content(&self) -> String {
-        let mut unit_file_content = String::new();
+        SECTION_ORDER
+            .iter()
+            .map(|section| self.sections.get_key_value(section))
+            .flatten()
+            .map(|(section, entries)| SystemDUnit::write_section(section, entries))
+            .collect::<Vec<_>>()
+            .join("\n\n")
+    }
 
-        // Iterate over all sections and write out its header and content
-        for (section, entries) in &self.sections {
-            unit_file_content.push_str(&format!("[{}]\n", section));
-            for (key, value) in entries {
-                unit_file_content.push_str(&format!("{}={}\n", key, value));
-            }
-            if section == SECTION_SERVICE {
-                // Add environment variables to Service section
-                for (name, value) in &self.environment {
-                    unit_file_content.push_str(&format!("Environment=\"{}={}\"\n", name, value));
-                }
-            }
-            unit_file_content.push('\n');
-        }
-        unit_file_content
+    fn write_section(section: &Section, entries: &HashMap<String, String>) -> String {
+        let header = format!("[{}]", section);
+
+        let mut body = entries
+            .iter()
+            .map(|(key, value)| format!("{}={}", key, value))
+            .collect::<Vec<_>>();
+        body.sort();
+
+        once(header).chain(body).collect::<Vec<_>>().join("\n")
```

Comment:
Would you mind changing this to `iter::once`?

https://doc.rust-lang.org/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html#creating-idiomatic-use-paths